src/pfe/file-watcher/server/test/functional-test/suites/generic/tests/workspace.ts

```diff
@@ -0,0 +1,299 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+import mocha from "mocha";
+import { expect } from "chai";
+import * as _ from "lodash";
+import path from "path";
+import fs from "fs";
+import { fail } from "assert";
+
+import * as genericLib from "../../../lib/generic";
+import { SocketIO } from "../../../lib/socket-io";
+
+import * as app_configs from "../../../configs/app.config";
+import * as eventConfigs from "../../../configs/event.config";
+import * as pfe_configs from "../../../configs/pfe.config";
+import * as timeoutConfigs from "../../../configs/timeout.config";
+
+export default class WorkspaceTest {
+    testName: string;
+
+    constructor(testName: string) {
+        this.testName = testName;
+    }
+
+    run(socket: SocketIO, runOnly?: boolean): void {
+        (runOnly ? describe.only : describe)(this.testName, () => {
+            const settingsPath = path.join(app_configs.codewindWorkspaceDir, ".config", "settings.json");
+            const backupSettingsPath = path.join(app_configs.codewindWorkspaceDir, ".config", "settings_bk.json");
+            const settingsContent = {
+                deploymentRegistry: "someregistry",
+                watcherChunkTimeout: 50000
+            };
+            this.afterEachHook(settingsPath, backupSettingsPath);
+            this.runReadWorkspaceSettings(socket, settingsPath, settingsContent, backupSettingsPath);
+            this.runDeploymentRegistryStatus(socket);
+            if (process.env.IN_K8) {
+                this.runTestDeploymentRegistry(settingsContent);
+            }
+        });
+    }
+
+    private afterEachHook(settingsPath: string, backupSettingsPath: string): void {
+        afterEach("delete backup settings file, revert back original file and load the current workspace settings", async () => {
+            if (await fs.existsSync(backupSettingsPath)) {
+                await fs.renameSync(backupSettingsPath, settingsPath);
+                await fs.chownSync(settingsPath, 1001, 2004);
+                await genericLib.readWorkspaceSettings();
```

Comment:
We can add expect or verification code here for `genericLib.readWorkspaceSettings();` to make sure the setting file is back to the original version correctly.